### ISSUE TITLE
When setting output strings that contain the value "25" the value isn't set as an output parameter.

### ISSUE THREAD

**Issue by vrfurl** [actions/runner#2760]
**Describe the bug**
I tried setting the output params with both $GITHUB_OUTPUT and  actions/github-script but every time I set a string value that contains `25` the string value is dropped.

**To Reproduce**
Ex Doesn't work:

```
      - name: Setting Outputs
        id: set_my_outputs
        uses: actions/github-script@v5
        with:
          script: |
            core.setOutput("DB_CONNECTION_URL", "25");
```

Or

```
      - name: Setting Outputs
        id: set_my_outputs
        run: |
            echo "Printing ENVs"
            env
            echo "DB_CONNECTION_URL=25" >> $GITHUB_OUTPUT
```

But this works:

Ex Works:

```
      - name: Setting Outputs
        id: set_my_outputs
        uses: actions/github-script@v5
        with:
          script: |
            core.setOutput("DB_CONNECTION_URL", "26");
```

Or

```
      - name: Setting Outputs
        id: set_my_outputs
        run: |
            echo "Printing ENVs"
            env
            echo "DB_CONNECTION_URL=26" >> $GITHUB_OUTPUT
```

In the example above the output param `DB_CONNECTION_URL` is empty but if I change the value to any other value that isn't 25 it works.  Note: The 25 doesn't need to be the only value in the string.  If I use something like: "mongodb+srv://commerce-pr-253-cluster.yqmub0q.mongodb.net" because 25 is present in the string it fails.  If remove the 25 it works. Very bizarre.

Any ideas on why this could be happening?

Thanks.

**Expected behavior**
We expected that we can set any string value.

## Runner Version and Platform

Version of your runner?

Current runner version: '2.307.1'

OS of the machine running the runner? OSX/Windows/Linux/...

Operating System
  Ubuntu
Runner Image
Image: ubuntu-20.04

## What's not working?
Please include error messages and screenshots.

When I attempt to access the output parameters that contain the value 25 in jobs that run after the one mentioned above the output parameter value is empty.

Using something like this which references all of the output params in the previous job:

```
steps:
      - name: All Output
        run: |
          echo "All Output values: ${{join(needs.deploy-mongodb-atlas.outputs.*, '-')}}"
```

yields empty results:

![image](https://github.com/actions/runner/assets/5599452/7890dcf2-a97d-4d7e-9f1e-019d1585e5a6)
****

However if I use a string value that doesn't contain the value 25 it works as expected:

ex:

![image](https://github.com/actions/runner/assets/5599452/a9d863a7-a3b7-49fa-9823-e0bf8049843c)

**Comment by dlucredativ** [comment]
Works for me. Can you post a complete workflow file? Mine is:

```
on:
  push:

jobs:
  one:
    runs-on: ubuntu-latest
    outputs:
      db_connection_url: ${{ steps.set_my_outputs.outputs.DB_CONNECTION_URL }}
    steps:
      - name: Setting Outputs
        id: set_my_outputs
        run: |
            echo "DB_CONNECTION_URL=25" >> $GITHUB_OUTPUT

  two:
    needs: one
    runs-on: ubuntu-latest
    steps:
      - run: |
          echo "All outputs: ${{ join(needs.one.outputs.*, '-') }}"
```
Result is
```
Run echo "All outputs: 25"
  echo "All outputs: 25"
  shell: /usr/bin/bash -e {0}
All outputs: 25
```

**Comment by vrfurl** [comment]
Hey @dlucredativ thank you for getting back to me.

So after you replied and showed that it worked for you I went through and commented out each line that I felt could be offending until I found the issue.

Not sure why this happens but we had two environment variables that were causing the issue above the `jobs` section:

ex:

```
env:
  APIGATEWAY_BURST_LIMIT: ${{secrets.APIGATEWAY_BURST_LIMIT}}
  APIGATEWAY_RATE_LIMIT: ${{secrets.APIGATEWAY_RATE_LIMIT}}
```

Whenever i commented those two secrets out then the number 25 started working.  Since they are Github Action secrets I can't read the value so I couldn't tell if there was some weird character in there or what not that was causing the issue.

To fix it I simply re-edited those two secrets and then it started working again.

Even though I'm happy that it's working now it's still pretty strange that an Environment variable set by a secret can cause the output parameters to stop working for the number `25` 🤯 

Thanks for your help, and I hope that Github can improve this in the future 🙏
